sklearn/metrics/tests/test_regression.py

```diff
@@ -330,3 +348,45 @@ def test_mean_absolute_percentage_error():
     y_true = random_number_generator.exponential(size=100)
     y_pred = 1.2 * y_true
     assert mean_absolute_percentage_error(y_true, y_pred) == pytest.approx(0.2)
+
+
+@pytest.mark.parametrize("distribution",
+                         ["normal", "lognormal", "exponential", "uniform"])
+@pytest.mark.parametrize("target_quantile", [0.05, 0.5, 0.75])
+def test_pinball_loss_on_constant_predictions(
+    distribution,
+    target_quantile
+):
+    if not hasattr(np, "quantile"):
+        pytest.skip("This test requires a more recent version of numpy "
+                    "with support for np.quantile.")
+
+    # Check that the pinball loss is minimized
+    n_samples = 100
+    rng = np.random.RandomState(42)
+    data = getattr(rng, distribution)(size=n_samples)
+
+    # Compute the best possible pinball loss for any constant predictor:
+    best_pred = np.quantile(data, target_quantile)
+    best_pred = np.full(n_samples, fill_value=best_pred)
+    best_pbl = pinball_loss(data, best_pred, alpha=target_quantile)
+
+    candidate_predictions = np.quantile(data, np.linspace(0, 1, 100))
+    for pred in candidate_predictions:
+        # Compute the pinball loss of a constant predictor:
+        constant_pred = np.full(n_samples, fill_value=pred)
+        pbl = pinball_loss(data, constant_pred, alpha=target_quantile)
+
+        # Check that the loss of this constant predictor is greater or equal
+        # than the loss of using the optimal quantile (up to machine
+        # precision):
+        assert pbl >= best_pbl - np.finfo(best_pbl.dtype).eps
```

Comment:
Could you modify this part and find the minimizing `y_pred` of the pinball loss by [scipy minimize](https://docs.scipy.org/doc/scipy-1.6.0/reference/generated/scipy.optimize.minimize.html) with method Nelder-Mead?
You test only on a grid: It could be that the optimal value is between 2 grid points.

The assert would then read
```python
assert numerical_minimum == pytest.approx(best_pbl)
assert numerical_argmin == pytest.approx(best_pred)
```

Reply:
I kept the grid to check the analytical formula on evenly spaces quantile values but I also added a call to the optimization procedure at the end.

If a regression is introduced, the grid is likely to fail first with a more informative error message. The optimization check is cherry on top.

Note that I had to increase the dataset size otherwise the solution is no unique enough.